cmd/piper.go

```diff
@@ -162,6 +164,65 @@ func PrepareConfig(cmd *cobra.Command, metadata *config.StepData, stepName strin
 	return nil
 }
 
+func convertTypes(config map[string]interface{}, options interface{}) map[string]interface{} {
+	typedOptions := reflect.ValueOf(options)
+	if typedOptions.Kind() == reflect.Ptr {
+		typedOptions = typedOptions.Elem()
+	}
+	optionsType := typedOptions.Type()
+
+	for key := range config {
+		// Find the field in options which will be mapped to 'key'
+		field := findField(key, typedOptions, optionsType)
+		if field == nil {
+			continue
+		}
+
+		entry := reflect.ValueOf(config[key])
+		if entry.Kind() != reflect.String {
+			// We can only convert from strings at the moment
+			continue
+		}
+		logWarning := true
+		switch field.Type.Kind() {
+		case reflect.String:
+			// Types match, ignore
+			logWarning = false
+		case reflect.Slice:
+			fallthrough
+		case reflect.Array:
+			if field.Type.Elem().Kind() == reflect.String {
+				config[key] = []string{entry.String()}
+				logWarning = false
+			}
+		case reflect.Bool:
+			value := strings.ToLower(entry.String())
+			if value == "true" {
+				config[key] = true
+				logWarning = false
+			} else if value == "false" {
+				config[key] = false
+				logWarning = false
+			}
+		}
+		if logWarning {
+			log.Entry().Warnf("Config value for '%s' is of unexpected type and is ignored", key)
+		}
+	}
+	return config
+}
+
+func findField(name string, typedOptions reflect.Value, optionsType reflect.Type) *reflect.StructField {
```

Comment:
IMO the naming typedOptions vs optionsType is somwhow confusing

Reply:
typedOptions is a value, optionsType is the type of the value.